src/vehicle_use.cpp

```diff
@@ -1744,6 +1744,136 @@ void vehicle::use_harness( int part, const tripoint &pos )
     }
 }
 
+std::vector<item_count_tuple> vehicle::get_comestible_liquids()
+{
+    std::vector<item_count_tuple> liquids;
+    std::map<item *, uint32_t> liquids_map;
+
+    liquids.clear();
+    liquids_map.clear();
+
+    for( auto &part : parts ) {
+        const int remaining = part.ammo_remaining();
+
+        if( !part.is_tank() || part.base.contents_made_of( SOLID ) || remaining <= 0 ) {
+            continue;
+        }
+
+        item *food_ptr = part.base.get_food();
+
+        if( food_ptr == nullptr || !food_ptr->is_comestible() ) {
+            continue;
+        }
+
+        const cata::value_ptr<islot_comestible> comestible = food_ptr->get_comestible();
+
+        if( comestible == nullptr || "DRINK" != comestible->comesttype ) {
+            continue;
+        }
+
+        const itype_id liquid_id = part.ammo_current();
+
+        if( liquid_id.empty() || liquid_id == "null" || liquid_id == "water" ) {
+            continue;
+        }
+
+        if( liquids_map.find( food_ptr ) == liquids_map.end() ) {
```

Comment:
This is not needed. Just add the count to the map (the below statement with `+=`). If the key does not exist in the map, it will be created and the value will be initialised (with 0 in this case).

Reply:
Fixed in upcoming commit.